### PR TITLE
Fixes MSSQL Limit in update query, closes #6636

### LINKED ISSUE #6636: MSSQL LIMIT in UPDATE
## What you are doing?

``` js
Account.update(request.body, {
    fields: [
      'nickname',
    ],
    limit: 1,
    where: {
      accountId: request.params.accountId
    }
  }).catch((error => {
    response.body = error;
    response.code = 500;
  }));
```
## What do you expect to happen?

The generated sql should have the MSSQL equivalent of a limit, in the form of a TOP

``` sql
"UPDATE TOP(1) [Account] SET [Nickname]=N'newNickname' OUTPUT INSERTED.* WHERE [Account_Id] = N'B7068730-59E1-4664-AAAB-A09D2CA10156'"
```
### Generated SQL

``` sql
"UPDATE [Account] SET [Nickname]=N'newNickname' OUTPUT INSERTED.* WHERE [Account_Id] = N'B7068730-59E1-4664-AAAB-A09D2CA10156' LIMIT 1"
```
### Error Text

`"Incorrect syntax near 'LIMIT'."`

I expect the UPDATE query to execute and update just one row, with the assurance that the query will update ONLY 1 row. In the example above my where clause is using the primary key but in other update queries I may be using a where clause which may bring back more than one row. In those instances I want to be able to use the limit option to ensure the query doesn't update beyond the rows I'm expecting to be updated. 

From what I can see in the source, the mssql dialect's version of [query-generator.js](https://github.com/sequelize/sequelize/blob/master/lib/dialects/mssql/query-generator.js) doesn't implement a method that overrides the behavior of the updateQuery method in [abstract/query-generator.js](https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js#L293), which is appending the LIMIT keyword to the end of the query. 

**Dialect:** mysql
**Database version:** 12.0.2000
**Sequelize version:** 4.0.0-0

### PR DESCRIPTION
**Pull request by harshithkashyap**
### Pull Request check-list
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

SQL Server uses TOP(n) keyword to limit the number of updated rows.
https://msdn.microsoft.com/en-us/library/ms177523.aspx

Closes #6636

### CODE REVIEW

File: test/unit/dialects/mssql/query-generator.test.js [hunk 1]
```diff
@@ -45,6 +45,46 @@ if (current.dialect.name === 'mssql') {
...
+      //Limit
+      expectsql(QueryGenerator.updateQuery('myTable', { foo: 'bar' }, {}, { limit: 1 }), {
+        mssql: "UPDATE TOP(1) [myTable] SET [foo]=N'bar' OUTPUT INSERTED.*"
+      });
+
+      expectsql(QueryGenerator.updateQuery('myTable', { foo: 'bar' }, { foo: 'value' }, { limit: 1 }), {
+        mssql: "UPDATE TOP(1) [myTable] SET [foo]=N'bar' OUTPUT INSERTED.* WHERE [foo] = N'value'"
+      });
+
+    });
```
Comment (mickhansen) [inline test/unit/dialects/mssql/query-generator.test.js]: the query-generator.test files are deprecated, please look at test/unit/sql instead

File: test/unit/sql/update.js [hunk 2]
```diff
@@ -34,5 +34,28 @@ describe(Support.getTestDialectTeaser('SQL'), function() {
...
+      }, {
+        timestamps: false
+      });
+
+      expectsql(sql.updateQuery(User.tableName, { username: 'new.username' }, { username: 'username' }, { limit: 1 }),
+      {
+        mssql: "UPDATE TOP(1) [Users] SET [username]=N'new.username' OUTPUT INSERTED.* WHERE [username] = N'username'",
+        postgres: "UPDATE \"Users\" SET \"username\"=\'new.username\' WHERE \"username\" = \'username\'",
+        sqlite: "UPDATE `Users` SET `username`='new.username' WHERE `username` = 'username' LIMIT 1",
+        default: "UPDATE `Users` SET `username`=\'new.username\' WHERE `username` = \'username\' LIMIT 1"
```
Comment (mickhansen) [inline test/unit/sql/update.js]: sqlite matches the default so no reason to have a seperate case, use `[]` like the other tests do to have dialect specfic quotesi n there.

**Comment by mickhansen** [comment]
Since this pretty much puts something in the front of the query, perhaps add the specific code to the mssql query generator, call the generic one from mssql and then replace `UPDATE` with `UPDATE TOP(${limit})`

File: test/unit/sql/update.js (hunk 2 above)
Comment (harshithkashyap) [inline test/unit/sql/update.js]: Right, I'll go ahead and remove sqlite. I suppose postgres is also not required as it doesn't support limit on update and would match the default case?
Sorry, didn't get the `[]` & using dialect specific quotes part. Do the expectation string go through processing before comparison as Sql Server needs `[]` and mysql needs back ticks?

Comment (janmeier) [inline test/unit/sql/update.js]: `default: "UPDATE [Users] SET [username]='new.username' WHERE [username] = 'username' LIMIT 1",`

[ and ] are then replaced with the quotes for that specific dialect (or kept as is for mssql :) )

Comment (mickhansen) [inline test/unit/sql/update.js]: IIRC postgres supports limit on update via a subquery.

File: lib/dialects/mssql/query-generator.js [hunk 3]
```diff
@@ -302,6 +302,15 @@ var QueryGenerator = {
     return generatedQuery;
   },
 
+  updateQuery(tableName, attrValueHash, where, options, attributes) {
+    let sql = AbstractQueryGenerator.updateQuery.call(this, tableName, attrValueHash, where, options, attributes);
```
Comment (felixfbecker) [inline lib/dialects/mssql/query-generator.js]: I think you should just be able to call `super.updateQuery()`

Comment (harshithkashyap) [inline lib/dialects/mssql/query-generator.js]: Yes, thanks :) Updated.

**Comment by harshithkashyap** [comment]
`npm run docs` seems to be broken. Is this expected on v4?

**Comment by janmeier** [comment]
Yep, docs are automagically generated by doclets for v4

**Comment by harshithkashyap** [comment]
@janmeier Any other changes needed?

File: lib/dialects/mssql/query-generator.js [hunk 4]
```diff
@@ -302,6 +302,15 @@ var QueryGenerator = {
     return generatedQuery;
   },
 
+  updateQuery(tableName, attrValueHash, where, options, attributes) {
+    let sql = super.updateQuery.call(this, tableName, attrValueHash, where, options, attributes);
```
Comment (felixfbecker) [inline lib/dialects/mssql/query-generator.js]: You can just do `super.updateQuery(tableName, ...)`

**Comment by harshithkashyap** [comment]
@janmeier Could you please merge this?

**Comment by janmeier** [comment]
Thanks!

**Pull request merged**